client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js

```diff
@@ -15,114 +15,108 @@ angular.module('App').controller(
         .then((durations) => {
           this.durations.available = durations;
           if (durations.length === 1) {
             [this.model.duration] = this.durations.available;
           }
-        }, (err) => {
+        })
+        .catch((err) => {
           this.$scope.resetAction();
           this.Alerter.alertFromSWS(this.$translate.instant('hosting_order_upgrade_error'), err, this.$scope.alerts.main);
-        }, (durations) => {
-          this.durations.available = durations;
         })
         .finally(() => {
           this.loading.durations = false;
         });
     }
 
-    /* Step 3 */
-    loadContracts() {
-      this.agree = false;
-      if (!this.model.duration.contracts || !this.model.duration.contracts.length) {
-        this.$rootScope.$broadcast('wizard-goToStep', 4);
-      }
-    }
+    formatTime(startTime) {
+      const time = moment(`T${startTime}`);
+      const now = moment(moment().format('TH:m'));
+      const isToday = time.diff(now) > 0;
 
-    /* Step 4 */
-    backToContracts() {
-      if (!this.model.duration.contracts || !this.model.duration.contracts.length) {
-        this.$rootScope.$broadcast('wizard-goToStep', 3);
-      }
+      return isToday
+        ? this.$translate.instant('hosting_order_upgrade_start_time_summary_today', { time: time.format('LT') })
+        : this.$translate.instant('hosting_order_upgrade_start_time_summary_tomorrow', { time: time.format('LT') });
     }
 
-    getResumePrice(price) {
+    formatPrice(price) {
       return price.value === 0 ? this.$translate.instant('price_free') : this.$translate.instant('price_ht_label', { price: price.text });
     }
 
     orderUpgrade() {
       this.loading.validation = true;
 
-      return this.Hosting.orderUpgrade(_.get(this.hosting, 'serviceName', this.$stateParams.productId), this.model.capacity, this.model.duration.duration)
-        .then((order) => {
-          this.Alerter.success(this.$translate.instant('hosting_order_upgrade_success', { t0: order.url, t1: order.orderId }), this.$scope.alerts.main);
-          this.atInternet.trackOrder({
-            name: `[hosting]::${this.model.capacity}[${this.model.capacity}]`,
-            page: 'web::payment-pending',
-            orderId: order.orderId,
-            priceTaxFree: order.prices.withoutTax.value,
-            price: order.prices.withTax.value,
-            status: 1,
-          });
-          this.$window.open(order.url, '_blank');
-        })
-        .catch((err) => {
-          this.Alerter.alertFromSWS(this.$translate.instant('hosting_order_upgrade_error'), err, this.$scope.alerts.main);
-        })
-        .finally(() => {
-          this.loading.validation = false;
-          this.$scope.resetAction();
+      const startTime = moment(`T${this.model.startTime}`).utc().format('HH:mm:ss');
+
+      return this.Hosting.orderUpgrade(
```

Comment:
```suggestion
      return this.Hosting
          .orderUpgrade(
              ...
```
would look nicer I think